docs/src/accessibility-testing-js.md

```diff
@@ -0,0 +1,292 @@
+
+## Example accessibility tests
+
+Accessibility tests work just like any other Playwright test. You can either create separate test cases for them, or integrate accessibility scans and assertions into your existing test cases.
+
+The following examples demonstrate a few basic accessibility testing scenarios.
+
+### Example 1: Scanning an entire page
+
+This example demonstrates how to test an entire page for automatically detectable accessibility violations. The test:
+1. Imports the `@axe-core/playwright` package
+2. Uses normal Playwright Test syntax to define a test case
+3. Uses normal Playwright syntax to navigate to the page under test
+4. Awaits `AxeBuilder.analyze()` to run the accessibility scan against the page
+5. Uses normal Playwright Test [expect] syntax to verify that there are no violations in the returned scan results
+
+```js tab=js-ts
+import { test, expect } from '@playwright/test';
+import AxeBuilder from '@axe-core/playwright'; // 1
+
+test.describe('homepage', function() { // 2
+  test('should not have any automatically detectable accessibility issues', async function({ page }) {
+    await page.goto('https://your-site.com/'); // 3
+
+    const accessibilityScanResults = await new AxeBuilder({ page }).analyze(); // 4
+
+    expect(accessibilityScanResults.violations).toEqual([]); // 5
+  });
+});
+```
+
+```js tab=js-js
+const { test, expect } = require('@playwright/test');
+const AxeBuilder = require('@axe-core/playwright').default; // 1
+
+test.describe('homepage', function() { // 2
+  test('should not have any automatically detectable accessibility issues', async function({ page }) {
+    await page.goto('https://your-site.com/'); // 3
+
+    const accessibilityScanResults = await new AxeBuilder({ page }).analyze(); // 4
+
+    expect(accessibilityScanResults.violations).toEqual([]); // 5
+  });
+});
+```
+
+### Example 2: Configuring axe to scan a specific part of a page
+
+`@axe-core/playwright` supports many configuration options for axe. You can specify these options by using a Builder pattern with the `AxeBuilder` class.
+
+For example, you can use [`AxeBuilder.include()`](https://github.com/dequelabs/axe-core-npm/blob/develop/packages/playwright/README.md#axebuilderincludeselector-string--string) to constrain an accessibility scan to only run against one specific part of a page.
+
+`AxeBuilder.analyze()` will scan the page *in its current state* when you call it. To scan parts of a page that are revealed based on UI interactions, use [Locators](./locators.md) to interact with the page before invoking `analyze()`:
+
+```js
+test('navigation menu flyout should not have automatically detectable accessibility violations', async function({ page }) {
+  await page.goto('https://your-site.com/');
+
+  await page.locator('button[aria-label="Navigation Menu"]').click();
+
+  await page.locator('#navigation-menu-flyout').waitFor();
```

Comment:
Let's note that it is important to `waitFor()` - otherwise we can run axe too early and it won't find the target element.